Sources/GoogleCloudProvider/Storage/StorageRequest.swift

```diff
@@ -18,48 +18,68 @@ extension HTTPHeaders {
 
 public class GoogleCloudStorageRequest {
     let refreshableToken: OAuthRefreshable
-    let httpClient: Client
     let project: String
 
+    let httpClient: Client
+    let responseDecoder: JSONDecoder
+    let dateFormatter: DateFormatter
+
     var currentToken: OAuthAccessToken?
     var tokenCreatedTime: Date?
     
     init(httpClient: Client, oauth: OAuthRefreshable, project: String) {
         self.refreshableToken = oauth
         self.httpClient = httpClient
         self.project = project
+        self.responseDecoder = JSONDecoder()
+        self.dateFormatter = DateFormatter()
+
+        self.dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        self.responseDecoder.dateDecodingStrategy = .formatted(self.dateFormatter)
     }
     
     func send<GCM: GoogleCloudModel>(method: HTTPMethod, headers: HTTPHeaders = [:], path: String, query: String, body: HTTPBody = HTTPBody()) throws -> Future<GCM> {
-        // if oauth token is not expired continue as normal
+        return try withToken({ token in
+            return try self.send(method: method, headers: headers, path: path, query: query, body: body, accessToken: token.accessToken).flatMap({ response in
+                return try self.responseDecoder.decode(GCM.self, from: response.http, maxSize: 65_536, on: response)
+            })
+        })
+    }
 
-        if let token = currentToken, let created = tokenCreatedTime, refreshableToken.isFresh(token: token, created: created) {
-            return try _send(method: method, headers: headers, path: path, query: query, body: body, accessToken: token.accessToken)
-        } else {
-            return try refreshableToken.refresh().flatMap({ (newToken) in
+    func send(method: HTTPMethod, headers: HTTPHeaders = [:], path: String, query: String, body: HTTPBody = HTTPBody()) throws -> Future<Data> {
```

Comment:
Maybe we can add a default param of type `GET` and it might be worth changing this method specifically to `get` since we can be confident that the return type is data and not a model of any kind. It would also clear up confusion of multiple send methods.

Reply:
Added the default param, however, I think that it's plausible we might use this method with other HTTP verbs in the future, so I hesitate to force it to `.GET`.

In terms of the multiple send methods - the return type disambiguates them.